### PR TITLE
Fix various issues with new multicluster feature

### PR DESCRIPTION
**Pull request by TheRealJon**
- Remove HOC patterns in Namespace and Project components that were causing issues with hooks when switching clusters
- useK8sGet hook was missing multicluster logic
- Clear stale redux state on error responses. Some redux actions and hooks were not clearing stale redux state on failed requests, which sometimes led to stale data being rendered.
- Disable prometheus via backend when multiple clusters are configured
- Fix golang pointer logic regression that was causing the same data to be used for every managed cluster config.
- Move ACM console link from the perspective switcher to the cluster switcher as "All Clusters" option
- Hide perspective switcher when ACM is the active perspective.
- Move active cluster redux state into the dynamic plugin SDK core redux state (also fixes a bug where resources were being created on the wrong cluster)
- Only save the last cluster to session storage instead of local storage/user settings. This partially addresses the issue where a user could get stuck being redirected to a cluster that is not accessible.

### CODE REVIEW

File: frontend/public/components/pod.tsx [hunk 1]
```diff
@@ -523,7 +523,7 @@ const getNetworkName = (result: PrometheusResult) =>
   result?.metric?.network_name || 'unnamed interface';
 
 // TODO update to use QueryBrowser for each graph
-const PodMetrics = requirePrometheus(({ obj }) => {
+const PodMetrics = ({ obj }) => {
```
Comment (jhadvig) [inline frontend/public/components/pod.tsx:526]:
```tsx
const PodMetrics: React.FC<{ obj: K8sResourceKind }> = ({ obj }) => {
```

File: frontend/public/locales/en/public.json [hunk 2]
```diff
@@ -543,6 +542,7 @@
   "(as a percentage of request)": "(as a percentage of request)",
   "{{name}} on pods": "{{name}} on pods",
   "{{name}} on<2></2>": "{{name}} on<2></2>",
+  "Metrics": "Metrics",
```
Comment (jhadvig) [inline frontend/public/locales/en/public.json:545]: I assume this change was autogenerated ?

**Review by jhadvig — COMMENTED** [review]
@TheRealJon the PR good 👍  … Adding two comments. The backend changes make sense, but would rather have extra pair of eyes on the frontend since Im not  aware of all the details.

**Review by spadgett — COMMENTED** [review]
Thanks @TheRealJon. Looks good overall. Let's make sure multi-cluster is not publicly exposed in the SDK as it is subject to change.

If the Home -> Overview page is not hidden when metrics is disabled, we should hide that as well.

**Comment by TheRealJon** [comment]
> If the Home -> Overview page is not hidden when metrics is disabled, we should hide that as well.

@spadgett Do we want to hide that entire nav item/page or just the metrics card on that page?

**Comment by spadgett** [comment]
/override ci/prow/e2e-gcp-console

Overriding the test failure since this it just our feature branch and the failure was a known flake. We will make sure this passes CI before merging into master.

**Review by spadgett — APPROVED** [review]
/lgtm

**Comment by spadgett** [comment]
/override ci/prow/e2e-gcp-console

Overriding the test failure since this it just our feature branch and the failure was a known flake. We will make sure this passes CI before merging into master.

**Comment by TheRealJon** [comment]
Closing as all of these changes are included in https://github.com/openshift/console/pull/10900

**Pull request closed without merging**